compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -670,10 +679,15 @@ static void PrintService(const ServiceDescriptor* service,
   (*vars)["service_name"] = service->name();
   (*vars)["file_name"] = service->file()->name();
   (*vars)["service_class_name"] = ServiceClassName(service);
+  #ifdef GRPC_VERSION
+    (*vars)["grpc_version"] = "(version "+string(XSTR(GRPC_VERSION))+")";
+  #else
```

Comment:
You shouldn't need the cast to string, and you shouldn't even need the `+`:
```
"(version " XSTR(GRPC_VERSION) ")"
```

Welcome to the magic of pre-processors and auto-string concatenation.